op-defender/psp_executor/README.md

```diff
@@ -8,28 +8,75 @@ The service is designed to listen on a port and execute a PSP onchain when a req
 
 ```shell
-go run ../cmd/defender psp_executor --privatekey XXXXXX --receiver.address 0xDEADBEEFDEADBEEFDEADBEEFDEADBEEFDEADBEEF --rpc.url http://localhost:8545 --port.api 8080
+go run ../cmd/defender psp_executor --privatekey 2a[..]c6 --safe.address 0x837DE453AD5F21E89771e3c06239d8236c0EFd5E --path /tmp/psps.json --chainid 11155111 --superchainconfig.address 0xC2Be75506d5724086DEB7245bd260Cc9753911Be --rpc.url http://localhost:8545 --port.api 8080
+```
+
+Explanation of the options:
+| Argument | Value | Explanation |
+| ---------------------------- | ------------------------------------------ | ------------------------------------ |
+| `--privatekey` | 2a[..]c6 | Private key for transaction signing |
+| `--safe.address` | 0x837DE453AD5F21E89771e3c06239d8236c0EFd5E | Address of the Safe contract that presigned the transation|
+| `--path` | /tmp/psps.json | Path to JSON file containing PSPs |
+| `--chainid` | 11155111 | Chain ID for the network |
+| `--superchainconfig.address` | 0xC2Be75506d5724086DEB7245bd260Cc9753911Be | Address of SuperchainConfig contract |
+| `--rpc.url` | http://localhost:8545 | URL of the RPC node |
+| `--port.api` | 8080 | Port for the HTTP API server |
+
+**PSPs Format**
+The PSPs are stored with a JSON format. The JSON file should contain an array of PSPs. Each PSP should have the following fields:
+
+```JSON
+[
+  {
+    "chain_id": "11155111",
+    "rpc_url": "https://ethereum-sepolia.publicnode.com",
+    "created_at": "2024-08-22T20:00:06+02:00",
+    "safe_addr": "0x837DE453AD5F21E89771e3c06239d8236c0EFd5E",
+    "safe_nonce": "0",
+    "target_addr": "0xfd7E6Ef1f6c9e4cC34F54065Bf8496cE41A4e2e8",
+    "script_name": "PresignPauseFromJson.s.sol",
+    "data": "[REDACTED]",
+    "signatures": [
+      {
+        "signer": "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266",
+        "signature": "[REDACTED]"
+      },
+      {
+        "signer": "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266",
+        "signature": "[REDACTED]"
+      }
+    ],
+    "calldata": "[REDACTED]"
+  }
+]
+
 ```
 
+The example above is starting by `[` and finishing by `]` as this is an array of PSPs. So here, only 1 PSP is defined.
+
 ### 2. Request the HTTP API
 
-To use the HTTP API you can use the following `curl` command with the following fields:
+![cde1a10c6f3eada39b2c27775faefd093623471db95bde4afcf06b9d43f3211b](https://github.com/user-attachments/assets/a75df38c-df06-4195-831b-08edb036a343)
 
-![image](https://github.com/user-attachments/assets/3edc2ee5-6dfd-4872-9bc6-e3ead7444a96)
+To use the HTTP API you can use the following `curl` command with the following fields:
 
 ```bash
-curl -X POST http://${HTTP_API_PSP}:${PORT}/api/psp_execution \-H "Content-Type: application/json" \-d '{"pause": true, "timestamp": 1719432011, "operator": "Tom"}'
+curl -X POST http://localhost:8080/api/psp_execution \-H "Content-Type: application/json" \-d '{"Pause":true,"Timestamp":1596240000,"Operator":"tom"}'
```

Comment:
Should Timestamp and Operator be set by the server?

Reply:
That's correct, and yes this is part of our current specs (for the analyzer and security oncall) authentication. 
No impact for us and this is currently desired.